src/twitch/twitch-eventsub-stream.ts

```diff
@@ -1,101 +1,54 @@
-import { Context, Effect, Layer } from "effect";
+import { Effect, Layer } from "effect";
 import {
   Message,
   MessagePubSub,
-  type CurrentlyPlayingRequestMessage,
-  type SongRequestMessage,
 } from "../pubsub/message-pubsub";
 import { TwitchConfig } from "./twitch-config";
 import { TwitchEventSubClient } from "./twitch-eventsub";
-import { Stream } from "../effect-exports";
 
-export type IChatMessageStream = Stream<CurrentlyPlayingRequestMessage>;
-
-const makeChatMessageStream = Effect.gen(function* () {
+const make = Effect.gen(function* () {
   const config = yield* TwitchConfig;
   const eventsub = yield* TwitchEventSubClient;
+  const pubsub = yield* MessagePubSub;
 
-  return Stream.async<CurrentlyPlayingRequestMessage>((emit) => {
-    eventsub.onChannelChatMessage(
+  yield* Effect.acquireRelease(
```

Comment:
Instead of going from event sub thing -> stream -> pubsub, you can just go straight into the pubsub using the unsafePublish api we added.

Using `Effect.acquireRelease` also means the subscription is cleaned up on shutdown.